src/core/ledger/ledger.test.js

```diff
@@ -847,21 +879,118 @@ describe("core/ledger/ledger", () => {
+            balance: g("17"),
+            paid: g("17"),
+            active: true,
+          };
+          expect(ledger.accounts()).toEqual([ac1, ac2]);
+        });
+
+        it("should record the allocations and distribution", () => {
+          expect(ledger.allocation(allocation1.id)).toEqual(allocation1);
+          expect(ledger.allocation(allocation2.id)).toEqual(allocation2);
+          expect(ledger.distribution(distribution.id)).toEqual(distribution);
+          expect(Array.from(ledger.allocations())).toEqual([
+            allocation1,
+            allocation2,
+          ]);
+          expect(Array.from(ledger.distributions())).toEqual([distribution]);
+          expect(ledger.distributionByAllocationId(allocation1.id)).toEqual(
+            distribution
+          );
+          expect(ledger.distributionByAllocationId(allocation2.id)).toEqual(
+            distribution
+          );
+        });
+      });
+
+      describe("when there are multiple distributions", () => {
+        let ledger;
+        const allocation1 = {
+          policy: {policyType: "IMMEDIATE", budget: g("10")},
+          id: uuid.random(),
+          receipts: [
+            {amount: g("3"), id: id1},
+            {amount: g("7"), id: id2},
+          ],
         };
-        const ac2 = {
-          identity: identity2(),
-          balance: g("17"),
-          paid: g("17"),
-          active: true,
+        const allocation2 = {
+          id: uuid.random(),
+          policy: {policyType: "BALANCED", budget: g("20")},
+          receipts: [
+            {amount: g("10"), id: id1},
+            {amount: g("10"), id: id2},
+          ],
+        };
+        const distribution1 = {
+          credTimestamp: 1,
+          allocations: [allocation1],
+          id: uuid.random(),
         };
-        expect(ledger.accounts()).toEqual([ac1, ac2]);
+        const distribution2 = {
+          credTimestamp: 2,
+          allocations: [allocation2],
+          id: uuid.random(),
+        };
+
+        beforeEach(() => {
+          ledger = ledgerWithActiveIdentities();
+          ledger.distributeGrain(distribution1);
+          ledger.distributeGrain(distribution2);
+        });
+
+        it("should record the payments in the accounts", () => {
+          const ac1 = {
+            identity: identity1(),
+            balance: g("13"),
+            paid: g("13"),
+            active: true,
+          };
+          const ac2 = {
+            identity: identity2(),
+            balance: g("17"),
+            paid: g("17"),
+            active: true,
+          };
+          expect(ledger.accounts()).toEqual([ac1, ac2]);
+        });
+
+        it("should record the allocations and distributions", () => {
+          expect(ledger.allocation(allocation1.id)).toEqual(allocation1);
+          expect(ledger.allocation(allocation2.id)).toEqual(allocation2);
+          expect(ledger.distribution(distribution1.id)).toEqual(distribution1);
```

Comment:
nit: may as well check for distribution 2.

Reply:
adding and merging